src/utils/asyncRequest.ts

```diff
@@ -125,7 +125,7 @@ export function asyncRequestIsFailed<R, E = void>(
 
 export function asyncRequestIsNotStarted<R, E = void>(
   request: AsyncRequest<R, E>,
-): boolean {
+): request is NotStarted {
```

Comment:
This is cool, we should maybe document this because type guards might be a little funky to read for people unfamiliar with them (like me)

Reply:
Yeah, where do you think would be the right place for that? We could link to the TS docs in the comment

Reply:
Maybe a typescript FAQ in c4cdocs? Not sure..